app/services/parameter_safe_render.rb

```diff
@@ -6,18 +6,40 @@ def initialize(host)
     @host = host
   end
 
+  # Faster method that DOES modify the input
+  def render!(value)
+    return value unless interpolate_erb?
+
+    render_object!(value)
+  end
+
+  # Slower method that does not modify the input
   def render(value)
+    return value unless interpolate_erb?
+
     render_object(value)
   end
 
   private
 
   attr_reader :host
 
-  def render_object(object)
-    return object unless (Setting[:interpolate_erb_in_parameters])
+  def interpolate_erb?
+    @interpolate_erb ||= Setting[:interpolate_erb_in_parameters]
```

Comment:
The caching doesn't work if the value is false.

Reply:
Good call, but it does work when it's true and it is by default. Shall I keep it this way?

Reply:
I lean towards putting it back (no memorizing), we only use it at a single place approx. four times.

Reply:
@tbrisker suggested to set the value in the initializer which I think is good enough.

Reply:
There is actually a way to memorize `false` values! Let me do it.